### ISSUE TITLE
jline 2.13 prints out weird characters

### ISSUE THREAD

**Issue by Minecrell** [jline/jline2#205]
When printing special characters like `\n` in jline 2.13 using a `ConsoleReader` it prints out weird characters like `^J` or `^M`. Has the usage changed compared to 2.12.1? When using jline 2.12.1 this code was working fine (I'm on Linux):

``` java
ConsoleReader reader = new ConsoleReader();
reader.println(ConsoleReader.RESET_LINE + "\n");
reader.flush();
```

With jline 2.12.1 this prints just an empty line correctly. However with jline 2.13 this prints the following:
`^M^J` instead of the empty line.

Any ideas?

**Comment by gnodet** [comment]
You can try ````reader.getOut().println("\r\n")```` if you want to print out raw characters.

**Comment by Minecrell** [comment]
Would that also work with printing colors on Windows? Before 2.13 this was working fine with the code I posted above.

**Comment by gnodet** [comment]
It should work with ansi sequences too.
However, it will bypass tabulations support, and may also have an undesirable behaviour, as the ConsoleReader does not know the cursor has moved.

If you really want to print a blank line, you can use ````reader.println()```` twice, that may be better.

**Comment by Minecrell** [comment]
The problem is, in the real application I have no control about the messages being printed because they're coming from the logging framework I'm using. Especially stack traces will always contain new lines without me able to really prevent them.

Additionally, colors doesn't seem to be printable in 2.13 directly either:

```java
ConsoleReader reader = new ConsoleReader();
reader.println(Ansi.ansi().a(Ansi.Attribute.RESET).fg(Ansi.Color.YELLOW).boldOff().toString() + "Test");
reader.flush();
```

Prints `^[[0;33;22mTest` on 2.13, it works fine if I print it using `reader.getOutput()`. Can it only print normal characters as of 2.13? In 2.12.1 this was all working fine.

**Comment by gnodet** [comment]
In 2.12.1, the only thing ````reader.println(xxx)```` was adding on top of ````reader.getOutput()```` is the support for tabs, so unless you actually need it, I suggest using the output writer directly.
I'll try to fix this regression somehow.

**Comment by sha-2** [comment]
Any updates in this?

**Comment by Minecrell** [comment]
@sha-2 Using `reader.getOutput()` instead worked fine for me, so using that now instead.

**Comment by darnir** [comment]
I'm currently using the `reader.getOutput()` workaround too, but would really like to see this regression fixed. 

Cannot use v1.12.1 where this works correctly because of a bug in Windows systems where non-English keyboards weren't supported well.

**Comment by bdw429s** [comment]
I think this is the same issue I am seeing just now after upgrading to JLine 2.14.1 to get the fix for #177
I used to send my own ANSI formatting via the reader.print() method for applying formatting or manipulating the cursor. Now all the escape codes are being printed to the console instead. This completely breaks my CLI app as all the formatting is gone in favor of the raw ANSI escape codes being printed to the screen.

**Comment by bdw429s** [comment]
Any update on this @gnodet ?  This is one of the regressions keeping CommandBox CLI from being able to upgrade, which sucks because there are some nice fixes I need in the newer versions of Jline.

**Comment by ctubbsii** [comment]
Ugh, this one affects all of Fedora, which ships with 2.13.

**Comment by gnodet** [comment]
Could you guys have a look at the recent commit ?
If you're all happy, I'll do a release of jline 2.15 asap.

**Comment by bdw429s** [comment]
@gnodet After a quick review in CommandBox, the snapshot of 2.15 seems to work well.  The regression is fixed and I'm finally able to use other fixes like https://github.com/jline/jline2/issues/177!

Thanks for your response to this.

**Comment by bdw429s** [comment]
Been testing more locally this week with the `2.15` snapshot and pleased to see I can also finally get the missing fixes for tickets like https://github.com/jline/jline2/issues/151 and https://github.com/jline/jline2/issues/150 that I've been waiting on.  Is there an ETA in mind for releasing `2.15` so I can plan accordingly?  I'm eyeing a conversion to Jline3, but if I could get `2.15` in my next release in the mean time that would be great.

**Comment by ctubbsii** [comment]
Any chance 2.15 will be released soon with this fix?

**Comment by gnodet** [comment]
I've released 2.14.3 in january.

**Comment by ctubbsii** [comment]
@gnodet Did 2.14.3 include this fix, then? I don't see this issue marked with any milestone.
